### PR TITLE
Exclude overlayfs module from main dracut package.

### PR DESCRIPTION
**Pull request by liulanze**
###### Merge Checklist  

**All** boxes should be checked before merging the PR *(just tick any boxes which don't apply to this PR)*
- [x] The toolchain has been rebuilt successfully (or no changes were made to it)
- [x] The toolchain/worker package manifests are up-to-date
- [x] Any updated packages successfully build (or no packages were changed)
- [x] Packages depending on static components modified in this PR (Golang, `*-static` subpackages, etc.) have had their `Release` tag incremented.
- [x] Package tests (%check section) have been verified with RUN_CHECK=y for existing SPEC files, or added to new SPEC files
- [x] All package sources are available
- [x] cgmanifest files are up-to-date and sorted (`./cgmanifest.json`, `./toolkit/scripts/toolchain/cgmanifest.json`, `.github/workflows/cgmanifest.json`)
- [x] LICENSE-MAP files are up-to-date (`./SPECS/LICENSES-AND-NOTICES/data/licenses.json`, `./SPECS/LICENSES-AND-NOTICES/LICENSES-MAP.md`, `./SPECS/LICENSES-AND-NOTICES/LICENSE-EXCEPTIONS.PHOTON`)
- [x] All source files have up-to-date hashes in the `*.signatures.json` files
- [x] `sudo make go-tidy-all` and `sudo make go-test-coverage` pass
- [x] Documentation has been updated to match any changes to the build system
- [x] Ready to merge

---

###### Summary 

What does the PR accomplish, why was it needed?

Exclude the dracut-overlayfs sub-package out of main dracut package to clear conflict caused by this sub-package.

###### Does this affect the toolchain?  

**NO**

###### Test Methodology

- Pipeline build id: buddybuild: https://dev.azure.com/mariner-org/mariner/_build/results?buildId=533662&view=results
- https://dev.azure.com/mariner-org/mariner/_build/results?buildId=533685&view=results

### CODE REVIEW

**Review by dmcilvaney — APPROVED** [review]

File: SPECS/dracut/dracut.spec [hunk 1]
```diff
@@ -156,6 +156,7 @@ ln -srv %{buildroot}%{_bindir}/%{name} %{buildroot}%{_sbindir}/%{name}
 %dir %{dracutlibdir}/modules.d
 %{dracutlibdir}/modules.d/*
 %exclude %{_libdir}/kernel
+%exclude %{_libdir}/dracut/modules.d/20overlayfs/*
```
Comment (cwize1) [inline SPECS/dracut/dracut.spec:159]: nit: Use `%{dracutlibdir}` instead of ` %{_libdir}/dracut` for the sake of consistency.

**Review by Camelron — APPROVED** [review]

File: SPECS/dracut/dracut.spec (hunk 1 above)
Comment (cwize1) [inline SPECS/dracut/dracut.spec:159]: Does this exclude the `20overlayfs`? Or does installing the package leave an empty directory? Should this be `%exclude %{_libdir}/dracut/modules.d/20overlayfs` instead?

Comment (liulanze) [reply SPECS/dracut/dracut.spec]: Fixed, thanks.

Comment (liulanze) [reply SPECS/dracut/dracut.spec]: Fixed, thanks.

**Review by Camelron — APPROVED** [review]

**Review by cwize1 — APPROVED** [review]

**Pull request merged**